packages/gatsby-plugin-manifest/src/gatsby-ssr.js

```diff
@@ -82,6 +83,31 @@ exports.onRenderBody = (
     }
   }
 
+  // Set page language if available
+  const { lang, start_url = `/` } = getPageLocalization(
+    pathname,
+    localize,
+    pluginOptions.lang
+  )
+  if (lang) {
+    setHtmlAttributes({ lang })
+  }
+
+  // Add metadata for localized page alternatives
+  if (Array.isArray(localize)) {
+    localize
+      .filter(localizedManifest => localizedManifest.lang !== lang)
+      .forEach(localizedManifest => {
+        headComponents.push(
+          <link
+            rel="alternate"
+            hrefLang={localizedManifest.lang}
+            href={pathname.replace(start_url, localizedManifest.start_url)}
+          />
```

Comment:
I wonder if we need to do some checks to make sure if there are no `alternate` links already set in case other plugin or user directly adds them?

This might result in duplicate sets of those tags.

Additionally - this makes some assumptions that won't always be true and there is currently no way to disable/control that feature. The assumption I'm most concerned about is the replacement of `start_url` with localized one - but it won't work for `/hello-world` / `/es/hola-mundo` cases (we would end up with `/es/hello-world` which potentially doesn't exist).

I do think this is valuable feature, we just should not enable this by default and instead have a way to opt into this and document requirements for this feature to work correctly.

What are your thoughts?

PS. We will want to bake in better support for i18n in the future so gatsby core would be more aware of page translations and could produce those `alternate` links, but we are not there yet

Reply:
Thank you for the valuable thoughts! I’ll try my best to make it an opt-in feature whenever I will have some spare time.